ci-automation/ci-config.env

```diff
@@ -129,3 +129,7 @@ AWS_PARALLEL="${PARALLEL_TESTS:-8}"
 : ${AZURE_VNET_SUBNET_NAME:="jenkins-vnet-westeurope"}
 AZURE_PARALLEL="${PARALLEL_TESTS:-20}"
 AZURE_LOCATION="${AZURE_LOCATION:-westeurope}"
+
+# -- Openstack --
+: ${OPENSTACK_IMAGE_NAME:='flatcar_production_openstack_image.img.gz'}
+OPENSTACK_PARALLEL="${PARALLEL_TESTS:-4}"
```

Comment:
What are possible numbers here, what are the constraints?

Reply:
```suggestion
# We have 10 VM slots and 3 VMs per test, so max 3 tests in parallel
OPENSTACK_PARALLEL="${PARALLEL_TESTS:-3}"
```
Ok, so we have 10 VM slots, or? Meaning that 3 is the better default because there are max 3 VMs per test.
For that to actually work without conflicts we would also have to use a lockable resource.

Reply:
Mm interesting. I gave a try with `--parallel 3` and `kubeadm.*` tests. The 5th instance creation has raised an error:
```
Traceback (most recent call last): File "/opt/stack/nova/nova/conductor/manager.py", line 1554, in schedule_and_build_instances host_lists = self._schedule_instances(context, request_specs[0], File "/opt/stack/nova/nova/conductor/manager.py", line 908, in _schedule_instances host_lists = self.query_client.select_destinations( File "/opt/stack/nova/nova/scheduler/client/query.py", line 41, in select_destinations return self.scheduler_rpcapi.select_destinations(context, spec_obj, File "/opt/stack/nova/nova/scheduler/rpcapi.py", line 160, in select_destinations return cctxt.call(ctxt, 'select_destinations', **msg_args) File "/usr/local/lib/python3.8/dist-packages/oslo_messaging/rpc/client.py", line 189, in call result = self.transport._send( File "/usr/local/lib/python3.8/dist-packages/oslo_messaging/transport.py", line 123, in _send return self._driver.send(target, ctxt, message, File "/usr/local/lib/python3.8/dist-packages/oslo_messaging/_drivers/amqpdriver.py", line 689, in send return self._send(target, ctxt, message, wait_for_reply, timeout, File "/usr/local/lib/python3.8/dist-packages/oslo_messaging/_drivers/amqpdriver.py", line 681, in _send raise result nova.exception_Remote.NoValidHost_Remote: No valid host was found. Traceback (most recent call last): File "/usr/local/lib/python3.8/dist-packages/oslo_messaging/rpc/server.py", line 241, in inner return func(*args, **kwargs) File "/opt/stack/nova/nova/scheduler/manager.py", line 209, in select_destinations raise exception.NoValidHost(reason="") nova.exception.NoValidHost: No valid host was found. 
```